### ISSUE TITLE
False positive: color-contrast rule fails when body height set to 0-8px

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#1456]
Color-contrast rule fails when body height is set to 0-8px, irrespective of the presence of sufficient background/foreground contrast ratio.

Demo: https://codepen.io/iamrafan/pen/WmLJmw?editors=1000#0

<pre><code>axe-core version: 3.1.2
axe-Coconut version: 3.7.0</code></pre>

**Comment by WilcoFiers** [comment]
That's pretty darn obscure, but jup, that's a false positive.

**Comment by scurker** [comment]
I did some digging to see if I could determine the root of the problem.

In the example, there's a link element of which we take the center x/y coordinates in order to determine the background element stack. It just so happens the y coordinate is _17 pixels_ which when we take into account the body height of `8px` and the user agent margin of `8px`, `body` ends up being left out of the background stack with `elementsFromPoint()`.

However, this is a case where the visible canvas has a special background color according to the spec:

[CSS3 Backgrounds Spec](https://www.w3.org/TR/css-backgrounds-3/#special-backgrounds):

> The document canvas is the infinite surface over which the document is rendered. Since no element corresponds to the canvas, in order to allow styling of the canvas CSS propagates the background of the root element (or, in the case of HTML, the `<body>` element) as described below. However, if no boxes are generated for the element whose background would be used for the canvas (for example, if the root element has display: none), then the canvas background is transparent.

Since the visible HTML canvas has a background color but not a intrinsically defined one that's within the coordinates, the rule fails. We should probably have `html` inherit its color value from `<body>` when no background-color exists for `html`.

**Comment by jeankaplansky** [comment]
No docs required.

**Comment by chandana7393** [comment]
Color-contrast violation is not throwing when height is set to 0-8px.
![color_contrast_height](https://user-images.githubusercontent.com/45285854/58779197-c5e1cc00-85f2-11e9-881b-ba1f5f0b3296.PNG)
**Tested Environment:**
Axe-coconut - 3.8.0.15404v
Chrome - 74v
OS - Windows 10 64 bit.
